dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusAArch64.java

```diff
@@ -13,22 +8,34 @@
 import com.dat3m.dartagnan.expression.type.IntegerType;
 import com.dat3m.dartagnan.expression.type.TypeFactory;
 import com.dat3m.dartagnan.parsers.LitmusAArch64BaseVisitor;
-import com.dat3m.dartagnan.parsers.LitmusAArch64Parser;
+import com.dat3m.dartagnan.parsers.LitmusAArch64Parser.*;
 import com.dat3m.dartagnan.parsers.program.utils.ProgramBuilder;
-import static com.dat3m.dartagnan.parsers.program.utils.ProgramBuilder.replaceZeroRegisters;
 import com.dat3m.dartagnan.program.Program;
 import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.Event;
 import com.dat3m.dartagnan.program.event.EventFactory;
-import com.dat3m.dartagnan.program.event.arch.StoreExclusive;
+import com.dat3m.dartagnan.program.event.arch.Xchg;
 import com.dat3m.dartagnan.program.event.core.Label;
-import com.dat3m.dartagnan.program.event.core.Load;
+import com.dat3m.dartagnan.program.event.metadata.CustomPrinting;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import java.math.BigInteger;
+import java.util.*;
+
+import static com.dat3m.dartagnan.parsers.program.utils.ProgramBuilder.replaceZeroRegisters;
+import static com.dat3m.dartagnan.program.event.Tag.ARMv8.*;
+import static com.google.common.base.Preconditions.checkArgument;
 
 public class VisitorLitmusAArch64 extends LitmusAArch64BaseVisitor<Object> {
 
-    private record CmpInstruction(Expression left, Expression right) {};
+    private record CmpInstruction(Expression left, Expression right) {}
+
     private final ProgramBuilder programBuilder = ProgramBuilder.forArch(Program.SourceLanguage.LITMUS, Arch.ARM8);
     private final TypeFactory types = programBuilder.getTypeFactory();
     private final IntegerType archType = types.getArchType();
+    private final IntegerType i8 = types.getIntegerType(8);
+    private final IntegerType i16 = types.getIntegerType(16);
+    private final IntegerType i32 = types.getIntegerType(32);
```

Comment:
Why not `i64`? Are you using archType instead? This could break if one changes `archType`

Reply:
Fine, this visitor is for AArch64, after all.  It probably should never adapt to alternative values for `archType`.